releasenotes/notes/deprecate-circuit-qasm-e5bbad027388e3f2.yaml

```
@@ -0,0 +1,5 @@
+---
+upgrade:
+  - |
+    Removed deprecated `Instruction.qasm` function. Use function :func:`qiskit.qasm2.dump()` or
+    function :func:`qiskit.qasm3.dump` instead.
```

The replacement for `Instruction.qasm` itself is purely `qiskit.qasm2.dump{,s}` on the containing circuit (not `qasm3`) - part of the reason we got rid of `QuantumCircuit.qasm` is because it doesn't include the version number.

As a minor note, the `` :func:`...` `` crossrefs in Sphinx only need to represent the object itself; whether it's written with a `()` following is controlled in the styling during display, so this release note might want to be something like:
```suggestion
    Removed the ``Instruction.qasm`` method, which was deprecated in Qiskit 0.45.0.
    Use :func:`qiskit.qasm2.dump` with a complete :class:`.QuantumCircuit` instead.
```